apps/services/user-profile/src/app/v2/user-profile.service.ts

```diff
@@ -34,37 +39,48 @@ export class UserProfileService {
   async findById(
     nationalId: string,
     useMaster = false,
+    clientType: ClientType = ClientType.THIRD_PARTY,
   ): Promise<UserProfileDto> {
     const userProfile = await this.userProfileModel.findOne({
       where: { nationalId },
       useMaster,
     })
 
     if (!userProfile) {
-      return {
-        nationalId,
-        email: null,
-        mobilePhoneNumber: null,
-        locale: null,
-        mobilePhoneNumberVerified: false,
-        emailVerified: false,
-        documentNotifications: true,
-        needsNudge: null,
-        emailNotifications: true,
-      }
+      return this.filterByClientTypeAndRestrictionDate(
+        clientType,
+        {
+          nationalId,
+          email: null,
+          mobilePhoneNumber: null,
+          locale: null,
+          mobilePhoneNumberVerified: false,
+          emailVerified: false,
+          documentNotifications: true,
+          needsNudge: null,
+          emailNotifications: true,
+          isRestricted: false,
+        },
+        null,
+      )
     }
 
-    return {
-      nationalId: userProfile.nationalId,
-      email: userProfile.email,
-      mobilePhoneNumber: userProfile.mobilePhoneNumber,
-      locale: userProfile.locale,
-      mobilePhoneNumberVerified: userProfile.mobilePhoneNumberVerified,
-      emailVerified: userProfile.emailVerified,
-      documentNotifications: userProfile.documentNotifications,
-      needsNudge: this.checkNeedsNudge(userProfile),
-      emailNotifications: userProfile.emailNotifications,
-    }
+    return this.filterByClientTypeAndRestrictionDate(
```

Comment:
We should just pass the user profile object to the function and let it map to the dto object along with the property filtering.